bot/handlers/sending_videos/compile_selected_clips_handler.py

```diff
@@ -31,50 +29,62 @@ def __init__(self, message: str) -> None:
             super().__init__(self.message)
 
     def get_commands(self) -> List[str]:
-        return ["polaczklipy", "concatclips", "pk"]
+        return ["połączklipy", "polaczklipy", "concatclips", "pk"]
+
+    async def is_any_validation_failed(self, message: Message) -> bool:
+        content = message.text.split()
+        if len(content) < 2:
+            await self._reply_invalid_args_count(message, get_invalid_args_count_message())
+            return True
+        return False
 
     async def _do_handle(self, message: Message) -> None:
         content = message.text.split()
 
-        if len(content) < 2:
+        try:
+            clip_numbers = [int(clip) for clip in content[1:]]
+        except ValueError:
             return await self._reply_invalid_args_count(message, get_invalid_args_count_message())
 
-        clip_names = content[1:]
-        selected_clips_data = await self.__get_selected_clips_data(clip_names, message.from_user.username, message)
+        user_clips = await DatabaseManager.get_saved_clips(message.from_user.id)
+
+        selected_clips = []
+        for clip_number in clip_numbers:
+            if 1 <= clip_number <= len(user_clips):
+                selected_clips.append(user_clips[clip_number - 1])
+            else:
+                return await self._reply_invalid_args_count(message, get_invalid_args_count_message())
 
-        if not selected_clips_data:
+        if not selected_clips:
             return await self.__reply_no_matching_clips_found(message)
 
         selected_segments = []
-        for clip_data, duration in selected_clips_data:
+        for clip in selected_clips:
             temp_file = tempfile.NamedTemporaryFile(delete=False, delete_on_close=False, suffix=".mp4")
-            temp_file.write(clip_data)
+            temp_file.write(clip.video_data)
             temp_file.close()
             selected_segments.append({
                 "video_path": temp_file.name,
                 "start": 0,
-                "end": duration,
+                "end": clip.duration,
             })
 
+        total_duration = sum(clip.duration for clip in selected_clips)
+        is_admin_or_moderator = await DatabaseManager.is_admin_or_moderator(message.from_user.id)
+
+        if not is_admin_or_moderator:
+            if total_duration > settings.MAX_CLIP_DURATION:
+                await message.answer(get_clip_time_message())
+                return
+            if len(selected_segments) > settings.MAX_CLIPS_PER_COMPILATION:
+                await message.answer(get_max_clips_exceeded_message())
+                return
```

Comment:
to tez mozna schowac ig

Reply:
nwm czy o to ci chodziło ale chyba zrobiłem XD